src/Utils/ProductGalleryUtils.php

```diff
@@ -0,0 +1,68 @@
+<?php
+namespace Automattic\WooCommerce\Blocks\Utils;
+
+/**
+ * Utility methods used for the Product Gallery block.
+ */
+class ProductGalleryUtils {
+
+	/**
+	 * When requesting a full-size image, this function may return an array with a single image.
+	 * However, when requesting a non-full-size image, it will always return an array with multiple images.
+	 * This distinction is based on the image size needed for rendering purposes:
+	 * - "Full" size is used for the main product featured image.
+	 * - Non-full sizes are used for rendering thumbnails.
+	 *
+	 * @param int    $post_id Post ID.
+	 * @param string $size Image size.
+	 * @param array  $attributes Attributes.
+	 * @return array
+	 */
+	public static function get_product_gallery_images( $post_id, $size = 'full', $attributes = array() ) {
+		$product_gallery_images = array();
+		$product                = wc_get_product( $post_id );
+
+		if ( $product ) {
+			// Main product featured image.
+			$featured_image_id = $product->get_image_id();
+			// All other product gallery images.
+			$product_gallery_image_ids = $product->get_gallery_image_ids();
+
+			// We don't want to show the same image twice, so we have to remove the featured image from the gallery if it's there.
+			$all_product_gallery_image_ids = array_unique(
```

Comment:
Out of curiosity. I haven't noticed the `$product->get_gallery_image_ids();` fetching the main image ever.

In that case, the `array_unique` would seems unnecessary.

Did you ever encounter this scenario when working on this?

Reply:
Good question! I noticed this when I worked on E2E tests.



![image](https://github.com/woocommerce/woocommerce-blocks/assets/4463174/4dc7b7d5-5932-4602-b27b-1eb0afedb00c)



In this scenario (with a product that has two images with the same ID), without the code above, this is the output on the frontend:

![image](https://github.com/woocommerce/woocommerce-blocks/assets/4463174/f8df2680-5a41-4d75-a0e5-027a5746c341)

This happens because the logic that renders the main image is based on the `selectedId` value. So, if there are two images with the same ID, both will be visible. Does it make sense?

Reply:
Right. That makes sense. Although I'll admit it's a but unlikely someone would intentionally add the same image twice (once as a "Featured Image" and the second time as a product gallery image).